src/shared/unit-file.c

```diff
@@ -568,21 +568,26 @@ int unit_file_build_name_map(
         /* Let's also put the names in the reverse db. */
         const char *dummy, *src;
         HASHMAP_FOREACH_KEY(dummy, src, ids) {
-                _cleanup_free_ char *inst = NULL, *dst_inst = NULL;
-                const char *dst;
+                _cleanup_free_ char *inst = NULL, *dst = NULL;
+                const char *dst_path;
 
-                r = unit_ids_map_get(ids, src, &dst);
+                r = unit_ids_map_get(ids, src, &dst_path);
                 if (r < 0)
                         continue;
 
-                if (null_or_empty_path(dst) != 0)
+                if (null_or_empty_path(dst_path) != 0)
                         continue;
 
-                dst = basename(dst);
+                r = path_extract_filename(dst_path, &dst);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to extract file name from %s, ignoring: %m", dst_path);
+                        continue;
+                }
 
                 /* If we have an symlink from an instance name to a template name, it is an alias just for
                  * this specific instance, foo@id.service ↔ template@id.service. */
                 if (unit_name_is_valid(dst, UNIT_NAME_TEMPLATE)) {
+                        _cleanup_free_ char *dst_inst = NULL;
```

Comment:
Please move the declaration to `if (t == UNIT_NAME_INSTANCE)` branch below.

Reply:
just two lines below, yes.. under `if (t < 0)` , that does makes sense in context.

@awilfox nearly there! :grinning: